substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -3169,8 +3229,34 @@ impl<T: Config> Pallet<T> {
 				"depositor must always have MinCreateBond stake in the pool, except for when the \
 				pool is being destroyed and the depositor is the last member",
 			);
+
+			let expected_tvl: BalanceOf<T> = BondedPools::<T>::iter()
```

Comment:
I think you are iterating inside [another loop](https://github.com/paritytech/polkadot-sdk/pull/1322/files#diff-fd1d9327e2dde740889545cb51e8e116c70b311d4318fd031cb95735ffdbb936L3152), probably mistake?

Also try to use the existing loop of BondedPools to calculate the TVL and add an ensure check afterwards.

Reply:
yes, indeed mistake. 👍 
I will change this as you suggest.

Reply:
What do you think about having them in separate iterations? The existing iterations over PoolMembers and BondedPools use fallible functions which could lead to having only some of the errors displayed, if the existing checks fail. I feel like the performance difference would be acceptable.